unit_tests/customizer/grasp_customization.cpp

```diff
@@ -0,0 +1,150 @@
+using namespace osrm::partition;
+using namespace osrm::util;
+
+namespace
+{
+struct MockEdge
+{
+    NodeID start;
+    NodeID target;
+    EdgeWeight weight;
+    bool operator<(const MockEdge &other)
+    {
+        return std::tie(start, target, weight) < std::tie(other.start, other.target, other.weight);
+    }
+    bool operator==(const MockEdge &other)
+    {
+        return std::tie(start, target, weight) == std::tie(other.start, other.target, other.weight);
+    }
+    bool operator!=(const MockEdge &other) { return !(*this == other); }
+};
+std::ostream &operator<<(std::ostream &out, const MockEdge &m)
+{
+    out << m.start << "->" << m.target << "(" << m.weight << ")";
+    return out;
+}
+
+auto makeGraph(const MultiLevelPartition &mlp, const std::vector<MockEdge> &mock_edges)
+{
+    struct EdgeData
+    {
+        EdgeWeight weight;
+        bool forward;
+        bool backward;
+    };
+    using Edge = static_graph_details::SortableEdgeWithData<EdgeData>;
+    std::vector<Edge> edges;
+    std::size_t max_id = 0;
+    for (const auto &m : mock_edges)
+    {
+        max_id = std::max<std::size_t>(max_id, std::max(m.start, m.target));
+        edges.push_back(Edge{m.start, m.target, m.weight, true, false});
+        edges.push_back(Edge{m.target, m.start, m.weight, false, true});
+    }
+    std::sort(edges.begin(), edges.end());
+    return partition::MultiLevelGraph<EdgeData, osrm::storage::Ownership::Container>(
+        mlp, max_id + 1, edges);
+}
+}
+
+BOOST_AUTO_TEST_SUITE(grasp_customization_tests)
+
+BOOST_AUTO_TEST_CASE(paper_test)
+{
+    /*
+        Example graph from paper:
+        Note: The paper marks 5 as border node, without having a border edge on level 1.
+              This is wrong hence I inserted an edge from 5 to 7 to make it true.
+        10
+          \
+           11---5---0---3---8---13
+           |   / \ /        |
+           6------1----     |
+           | /         \    |
+           7--------2---4---9---14
+                    |
+                   12
+    */
+    // node:                0  1  2  3  4  5  6  7  8  9 10 11 12 13 14
+    std::vector<CellID> l1{{0, 0, 1, 3, 2, 0, 0, 1, 3, 2, 0, 0, 1, 3, 2}};
+    std::vector<CellID> l2{{0, 0, 0, 1, 1, 0, 0, 0, 1, 1, 0, 0, 0, 1, 1}};
+    std::vector<CellID> l3{{0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0}};
+    MultiLevelPartition mlp{{l1, l2, l3}, {4, 2, 1}};
+
+    std::vector<MockEdge> edges = {{0, 1, 1},
+                                   {0, 3, 1},
+                                   {0, 5, 1},
+                                   {1, 4, 1},
+                                   {1, 5, 1},
+                                   {1, 6, 1},
+                                   {2, 4, 1},
+                                   {2, 7, 1},
+                                   {2, 12, 1},
+                                   {3, 8, 1},
+                                   {4, 9, 1},
+                                   {5, 7, 1},
```

Comment:
The directed graph with the edge  is ![m](https://cloud.githubusercontent.com/assets/4421046/24860371/8d4515d6-1df4-11e7-8489-2693a2cf98df.png) and the corresponding downwards graph is 
![s](https://cloud.githubusercontent.com/assets/4421046/24860504/314e8e00-1df5-11e7-87fc-4ae86b9df932.png)

I would suggest to add another test without the edge and with bidirectional edges, so the graph will
![nm](https://cloud.githubusercontent.com/assets/4421046/24860532/4e7219de-1df5-11e7-93a3-f7cbb7a3b695.png) and the downwards graphs 
![ns](https://cloud.githubusercontent.com/assets/4421046/24860543/5b7c5004-1df5-11e7-989a-8c9b701ba626.png)